examples/user_guide/Performance_and_Debugging.ipynb

```diff
@@ -46,6 +46,80 @@
     "The first time the app is loaded the data will be cached and subsequent sessions will simply look up the data in the cache, speeding up the process of rendering. If you want to warm up the cache before the first user visits the application you can also provide the `--warm` argument to the `panel serve` command, which will ensure the application is initialized once on launch."
    ]
   },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Threading and Async IO\n",
+    "\n",
+    "Threading and async are both approaches to speed up processing in Python using concurrency. Since we can't provide a complete primer on either threading or asynchronous processing here, if you are not familiar with these concepts we recommend reading up on them before continuing. Read about [threading in Python here](https://realpython.com/intro-to-python-threading/) and [AsyncIO here](https://realpython.com/async-io-python/).\n",
+    "\n",
+    "When to use which approach cannot be answered easily and is never completely clear cut. As a general guide however use `asyncio` when you are performing many slow IO bound operations concurrently (on the order of 100 and more) and use threading when you are performing a limited number (less than the number of available threads). In some situations threading can also be useful for CPU intensive operations where the operation being executed [releases the GIL](https://realpython.com/async-io-python/), this includes many NumPy, Pandas and Numba functions.\n",
+    "\n",
+    "### Threading\n",
+    "\n",
+    "Threading in Panel can be enabled easily by setting the `config.nthreads` parameter. This will start a `ThreadPoolExecutor` with the specified number of threads (or if set to `0` it will set the number of threads based on your system, i.e. `min(32, os.cpu_count() + 4)`). Whenever an event is generated or a periodic callback fires Panel will then automatically dispatch the event to the executor. If launching an application with `panel serve` you can configure this option with the `--num-threads` option.\n",
+    "\n",
+    "### AsyncIO\n",
+    "\n",
+    "When using Python>=3.8 you can use async callbacks wherever you would ordinarily use a regular synchronous function. For instance you can use `pn.bind` on an async function:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import aiohttp\n",
+    "\n",
+    "widget = pn.widgets.IntSlider(start=0, end=10)\n",
+    "\n",
+    "async def get_img(index):\n",
+    "    async with aiohttp.ClientSession() as session:\n",
+    "        async with session.get(f\"https://picsum.photos/800/300?image={index}\") as resp:\n",
+    "            return pn.pane.JPG(await resp.read())\n",
+    "            \n",
+    "pn.Column(widget, pn.bind(get_img, widget))"
```

Comment:
I could not get this example to work, either in a notebook or as a script. Running the script showed this warning:
```
RuntimeWarning: coroutine 'get_img' was never awaited
  return f(self, obj, val)
```
I used python 3.8.

Reply:
Again, requires master.